framework/src/meshgenerators/XYDelaunayGenerator.C

```diff
@@ -293,7 +312,9 @@ XYDelaunayGenerator::generate()
   if (_smooth_tri || _output_subdomain_id)
     for (auto elem : mesh->element_ptr_range())
     {
-      mooseAssert(elem->type() == TRI3, "Unexpected non-Tri3 found in triangulation");
+      mooseAssert(elem->type() ==
+                      (_tri_elem_type == "TRI6" ? TRI6 : (_tri_elem_type == "TRI7" ? TRI7 : TRI3)),
+                  "Unexpected non-Tri found in triangulation");
```

Comment:
We no longer can fail just on non-Tri elements here, so we should probably spit more information out.

Something like:

```suggestion
                  "Unexpected element type " << Utility::enum_to_string(elem->type()) << " found in triangulation");
```

(with an `#include <libmesh/utility.h>` above to support it), perhaps?